onnxruntime/core/graph/graph_flatbuffers_utils.cc

```diff
@@ -196,8 +197,25 @@ Status LoadInitializerOrtFormat(const fbs::Tensor& fbs_tensor,
     const auto* fbs_raw_data = fbs_tensor.raw_data();
     ORT_RETURN_IF(nullptr == fbs_raw_data, "Missing raw data for initializer. Invalid ORT format model.");
 
-    // fbs_raw_data is uint8_t vector, so the size is byte size
-    initializer.set_raw_data(fbs_raw_data->Data(), fbs_raw_data->size());
+    if (can_use_flatbuffer_for_initializers && fbs_raw_data->size() > 127) {
+      initializer.set_data_location(ONNX_NAMESPACE::TensorProto_DataLocation_EXTERNAL);
+
+      const void* data_offset = fbs_raw_data->Data();
+      auto offset = gsl::narrow<ExternalDataInfo::OFFSET_TYPE>(reinterpret_cast<uintptr_t>(data_offset));
```

Comment:
should we use intptr_t here since OFFSET_TYPE is signed? also should we do a check like `static_assert(sizeof(void*) <= sizeof(ExternalDataInfo::OFFSET_TYPE))` somewhere?

Reply:
I used uintptr_t so that the gsl::narrow can pickup if we can't convert to a signed value. 

I can add the static_assert.

Reply:
> I used uintptr_t so that the gsl::narrow can pickup if we can't convert to a signed value.

wouldn't this limit the range of allowed pointer values? e.g., if signed OFFSET_TYPE and unsigned uintptr_t are the same width, only half of the uintptr_t range would be valid

Reply:
given the large range of an int64, it's probably fine practically and I don't have a strong opinion about it.
my thinking was that since we are going between void* and signed integer, intptr_t seems most appropriate.